spec/services/teachers/set_funding_eligibilty_spec.rb

```diff
@@ -0,0 +1,110 @@
+RSpec.describe Teachers::SetFundingEligibilty do
+  let(:teacher) { FactoryBot.create(:teacher) }
+  let(:author) { Events::SystemAuthor.new }
+  let(:service) { described_class.new(teacher:, author:) }
+
+  describe "#set!" do
+    context "when teacher is eligible for ECT training" do
+      before do
+        FactoryBot.create(:induction_period, :ongoing, teacher:)
+        FactoryBot.create(:ect_at_school_period, :ongoing, teacher:)
+      end
+
+      it "sets `first_became_eligible_for_ect_training_at` if not already set" do
+        expect { service.set! }.to change(teacher, :first_became_eligible_for_ect_training_at).from(nil)
```

Comment:
Can we expect it to `be_within(5.seconds).of(Time.zone.now)` on the tests that check the attributes are set?